### PR TITLE
Non impact full Plugin updates

### PR DESCRIPTION
**Pull request by cfkoehler**
## Before:
```
mvn versions:display-plugin-updates:
[INFO] The following plugin updates are available:
[INFO]   com.github.ekryd.sortpom:sortpom-maven-plugin ..... 2.15.0 -> 3.2.1
[INFO]   io.github.git-commit-id:git-commit-id-maven-plugin . 4.9.9 -> 5.0.0
[INFO]   maven-assembly-plugin .............................. 3.3.0 -> 3.4.2
[INFO]   maven-checkstyle-plugin ............................ 3.1.1 -> 3.2.1
[INFO]   maven-dependency-plugin ............................ 3.3.0 -> 3.5.0
[INFO]   maven-deploy-plugin ............................. 3.0.0-M1 -> 3.0.0
[INFO]   maven-enforcer-plugin ........................... 3.0.0-M2 -> 3.2.1
[INFO]   maven-failsafe-plugin ........................ 3.0.0-M3 -> 3.0.0-M8
[INFO]   maven-install-plugin ............................ 3.0.0-M1 -> 3.1.0
[INFO]   maven-jar-plugin ................................... 3.2.2 -> 3.3.0
[INFO]   maven-javadoc-plugin ............................... 3.4.0 -> 3.4.1
[INFO]   maven-jxr-plugin ................................... 3.2.0 -> 3.3.0
[INFO]   maven-pmd-plugin ................................. 3.16.0 -> 3.20.0
[INFO]   maven-project-info-reports-plugin .................. 3.3.0 -> 3.4.2
[INFO]   maven-release-plugin ............................ 2.5.3 -> 3.0.0-M7
[INFO]   maven-resources-plugin ............................. 3.2.0 -> 3.3.0
[INFO]   maven-site-plugin .............................. 3.11.0 -> 4.0.0-M4
[INFO]   maven-source-plugin ................................ 3.2.0 -> 3.2.1
[INFO]   maven-surefire-plugin ........................ 3.0.0-M3 -> 3.0.0-M8
[INFO]   maven-surefire-report-plugin ................. 3.0.0-M3 -> 3.0.0-M8
[INFO]   net.revelc.code.formatter:formatter-maven-plugin . 2.11.0 -> 2.21.0
[INFO]   org.codehaus.mojo:exec-maven-plugin ................ 3.0.0 -> 3.1.0
[INFO]   org.codehaus.mojo:versions-maven-plugin .......... 2.10.0 -> 2.14.2
[INFO]   org.sonarsource.scanner.maven:sonar-maven-plugin  3.7.0.1746 -> 3.9.1.2184
```

## After:
```
mvn versions:display-plugin-updates:
[INFO]   com.github.ekryd.sortpom:sortpom-maven-plugin ..... 2.15.0 -> 3.2.1
[INFO]   io.github.git-commit-id:git-commit-id-maven-plugin . 4.9.9 -> 5.0.0
[INFO]   net.revelc.code.formatter:formatter-maven-plugin . 2.11.0 -> 2.21.0
[INFO]   org.codehaus.mojo:versions-maven-plugin .......... 2.10.0 -> 2.14.2
```

Formatter is going to be updated in a seperate PR: https://github.com/NationalSecurityAgency/emissary/pull/406

# Couple of tasks:
- [x] Revert git plugin update (Not compatible with java 8 build environment)
- [x] Verify all versions are available in our build repos

### CODE REVIEW

**Review by ldhardy — DISMISSED** [review]

**Comment by cfkoehler** [comment]
Couple of tasks:

- [ ] Revert git plugin update (Not compatible with java 8 build environment)
- [ ] Verify all versions are available in our build repos

**Comment by cfkoehler** [comment]
@ldhardy removed the git and the sort plugin updates. I think this should be good to go now with our current jdk version

**Review by fbruton — APPROVED** [review]

**Comment by cfkoehler** [comment]
I think this is good except for checking all the versions are available. I will plan to do this 3/27/23

**Review by ldhardy — CHANGES_REQUESTED** [review]
I'll run the build on this once rebase happens.  I think if I approve it before you rebase it will mark the approval stale?  or drop it?  not sure.

**Review by ldhardy — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
     <dep.junit.version>4.13.2</dep.junit.version>
     <dep.logback.version>1.2.11</dep.logback.version>
     <!-- seems to have an impact with java 11 integration, update with caution -->
-    <dep.maven-surefire-plugin.version>3.0.0-M3</dep.maven-surefire-plugin.version>
+    <dep.maven-surefire-plugin.version>3.0.0-M8</dep.maven-surefire-plugin.version>
```
Comment (jpdahlke) [inline pom.xml:58]: 😬

Comment (cfkoehler) [reply pom.xml:58]: :(

Comment (cfkoehler) [reply pom.xml:58]: In my defense that comment came after I created this PR back in January :)

File: pom.xml [hunk 2]
```diff
@@ -921,15 +921,15 @@
...
+          <version>3.4.2</version>
           <configuration>
             <dependencyDetailsEnabled>false</dependencyDetailsEnabled>
           </configuration>
         </plugin>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-release-plugin</artifactId>
-          <version>2.5.3</version>
+          <version>3.0.0-M7</version>
```
Comment (jpdahlke) [inline pom.xml:932]:
```suggestion
          <version>3.0.0</version>
```
released in march, check for availability

File: pom.xml [hunk 3]
```diff
@@ -954,7 +954,7 @@
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-site-plugin</artifactId>
-          <version>3.11.0</version>
+          <version>4.0.0-M4</version>
```
Comment (jpdahlke) [inline pom.xml:957]:
```suggestion
          <version>3.12.1</version>
```
check for availability

**Review by jpdahlke — CHANGES_REQUESTED** [review]

**Comment by jpdahlke** [comment]
Can you also verify/bump the maven-deploy-plugin to 3.1.0, update the minimum maven version to 3.5.0 instead of 3.3.9 (impsort requires 3.5.0) and update the README to specify maven 3.5+ instead of 3.6+?

File: pom.xml (hunk 3 above)
Comment (cfkoehler) [reply pom.xml:957]: Confirmed availability

**Review by dev-mlb — APPROVED** [review]

**Pull request merged**